pkg/lib/authn/identity/service/service.go

```diff
@@ -395,6 +395,26 @@ func (s *Service) CheckDuplicated(is *identity.Info) (dupeIdentity *identity.Inf
 	return
 }
 
+func (s *Service) GetLoginIDDisplayID(userID string, loginIDKey string) (displayID string, err error) {
+	var loginIDs []*loginid.Identity
+
+	if userID != "" {
+		loginIDs, err = s.LoginID.List(userID)
+		if err != nil {
+			return
+		}
+		// No need to consider anonymous identity
+	}
+
+	for _, iden := range loginIDs {
+		if iden.LoginIDKey == loginIDKey {
+			displayID = loginIDToIdentityInfo(iden).DisplayID()
+		}
+	}
+
+	return
+}
+
```

Comment:
Actually no need to add this function. We should reuse the existing `func (s *Service) Get(userID string, typ authn.IdentityType, id string) (*identity.Info, error)`.

userID: You got it already.
typ: authn.IdentityTypeLoginID
id: the value of r.Form.Get("x_identity_id")

Of course we need to handle the identity not found error.